coderd/debug.go

```diff
@@ -107,6 +112,104 @@ func formatHealthcheck(ctx context.Context, rw http.ResponseWriter, r *http.Requ
 	}
 }
 
+// @Summary Get health settings
+// @ID get-health-settings
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Debug
+// @Success 200 {object} codersdk.HealthSettings
+// @Router /debug/health/settings [get]
+func (api *API) deploymentHealthSettings(rw http.ResponseWriter, r *http.Request) {
+	settingsJSON, err := api.Database.GetHealthSettings(r.Context())
+	if err != nil {
+		httpapi.Write(r.Context(), rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to fetch health settings.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	var settings codersdk.HealthSettings
+	err = json.Unmarshal([]byte(settingsJSON), &settings)
+	if err != nil {
+		httpapi.Write(r.Context(), rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to unmarshal health settings.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	if len(settings.DismissedHealthchecks) == 0 {
+		settings.DismissedHealthchecks = []string{}
+	}
+
+	httpapi.Write(r.Context(), rw, http.StatusOK, settings)
+}
+
+// @Summary Update health settings
+// @ID update-health-settings
+// @Security CoderSessionToken
+// @Accept json
+// @Produce json
+// @Tags Debug
+// @Param request body codersdk.UpdateHealthSettings true "Update health settings"
+// @Success 200 {object} codersdk.UpdateHealthSettings
+// @Router /debug/health/settings [put]
+func (api *API) putDeploymentHealthSettings(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	if !api.Authorize(r, rbac.ActionUpdate, rbac.ResourceDeploymentValues) {
+		httpapi.Write(ctx, rw, http.StatusForbidden, codersdk.Response{
+			Message: "Insufficient permissions to update health settings.",
+		})
+		return
+	}
+
+	var settings codersdk.HealthSettings
+	if !httpapi.Read(ctx, rw, r, &settings) {
+		return
+	}
+
+	err := validateHealthSettings(settings)
+	if err != nil {
+		httpapi.Write(r.Context(), rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to validate health settings.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	settingsJSON, err := json.Marshal(&settings)
+	if err != nil {
+		httpapi.Write(r.Context(), rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to marshal health settings.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	err = api.Database.UpsertHealthSettings(ctx, string(settingsJSON))
```

Comment:
nit: For PUT idempotency, should we check if existing == to be updated and return 304 if so?

Reply:
Yeah, that's why I was thinking about POST too :) I can adapt this logic 👍

Reply:
Fixed